integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,461 @@
+	cmd := exec.Command("kubectl", "get", "pods", "-n", app.name)
+	out, er := cmd.CombinedOutput()
+	if er != nil {
+		err = er
+		return
+	}
+	str := string(out)
+	splits := strings.Split(str, "\n")
+	whitespaces := regexp.MustCompile(`\s+`)
+	var res []podEntry
+	for i, line := range splits {
+		switch i {
+		case 0:
+			continue
+		default:
+			parts := whitespaces.Split(line, AllSubstrings)
+			if len(parts) == 5 {
+				podEntry := podEntry{parts[0], parts[1], parts[2], parts[3], parts[4]}
+				res = append(res, podEntry)
+			}
+		}
+	}
+	return res, nil
+}
+
+func checkStatusIs(app app, status string) (res string, err error) {
+	for {
+		appStatus, er := getStatus(app)
+		if er != nil {
+			err = er
+			return
+		}
+		allSame := true
+		for _, entry := range appStatus.streamletPods {
+			allSame = allSame && entry.status == status
+			if !allSame {
+				fmt.Printf("Entry [%s, %s] is not compliant with status [%s]", entry.streamlet, entry.status, status)
+				break
+			}
+		}
+		if allSame {
+			return status, nil
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func getStreamlets(appStatus appStatus) []string {
+	var res []string
+	for _, entry := range appStatus.streamletPods {
+		res = append(res, entry.streamlet)
+	}
+	return res
+}
+
+func getStatus(app app) (status appStatus, err error) {
+	cmd := exec.Command("kubectl", "cloudflow", "status", app.name)
+	out, er := cmd.CombinedOutput()
+	if er != nil {
+		err = er
+		return
+	}
+	str := string(out)
+	splits := strings.Split(str, "\n")
+
+	names := [5]string{"Name:", "Namespace:", "Version:", "Created:", "Status:"}
+	fields := [5]*string{&status.name, &status.namespace, &status.version, &status.created, &status.status}
+	var streamletPods []streamletPod
+
+	for i, line := range splits {
+
+		switch i {
+		case 0, 1, 2, 3, 4:
+			value, er := parseLineInN(line, 2)
+			if er != nil {
+				err = er
+				return
+			}
+			if value[0] != names[i] {
+				err = fmt.Errorf("unexpected header name. Got [%s] but expected [%s]", value[0], names[i])
+				return
+			}
+			*fields[i] = value[1]
+		case 5:
+			continue // separator line
+		case 6:
+			continue // titles
+		default:
+			if len(strings.TrimSpace(line)) == 0 {
+				continue
+			}
+			parts, er := parseLineInN(line, 5)
+			if er != nil {
+				err = er
+				return
+			}
+			var streamletPod streamletPod
+			streamletPod.streamlet = parts[0]
+			streamletPod.pod = parts[1]
+			streamletPod.status = parts[2]
+			restarts, er := strconv.Atoi(parts[3])
+			if er != nil {
+				err = er
+				return
+			}
+			streamletPod.restarts = restarts
+			ready, er := strconv.ParseBool(parts[4])
+			if er != nil {
+				err = er
+				return
+			}
+			streamletPod.ready = ready
+			streamletPods = append(streamletPods, streamletPod)
+		}
+	}
+	status.streamletPods = streamletPods
+	return status, nil
+}
+
+func parseLineInN(str string, segments int) (parsed []string, err error) {
+	parts := Whitespaces.Split(str, AllSubstrings)
```

Comment:
I would just move the regexp down here instead of having an exported global variable.